app/middleware/get-service-and-gateway-account.middleware.js

```diff
@@ -34,16 +33,11 @@ async function getGatewayAccountByExternalId (gatewayAccountExternalId, correlat
   } catch (err) {
     const logContext = {}
     logContext['error'] = err.message
+    logContext['error_code'] = err.errorCode
     logContext['GATEWAY_ACCOUNT_EXTERNAL_ID'] = gatewayAccountExternalId
     logContext[keys.CORRELATION_ID] = correlationId
 
-    if (err.errorCode === 404) {
```

Comment:
If it's a 404, do we not want to return an error? We probably want to continue and show the not authorised page if we're not explicitly returning a not found error anymore. So calling `next` without arguments is probably needed for the case of a 404.

Reply:
I misread, we aren't throwing an error in this case. Although we probably don't want to log at error level for a 404.